src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -73,14 +73,22 @@ def create_webapp(resource_group_name, name,
         result = create_app_service_plan(resource_group_name, plan, is_linux,
                                          sku or 'S1', number_of_workers or 1, location)
         plan_id = result.id
-
     webapp_def = Site(server_farm_id=plan_id, location=location)
     poller = client.web_apps.create_or_update(resource_group_name, name, webapp_def)
-    return AppServiceLongRunningOperation()(poller)
+    site = AppServiceLongRunningOperation()(poller)
+    profiles = list_publish_profiles(resource_group_name, name, None)
+    publishURL = profiles[1]["publishUrl"]                                            
```

Comment:
This is not reliable, suggest use comprehension like `next((for p in profiles if p['publishMethod'] == 'ftp'), None)`?.
Again, create a common routine so to share between both methods